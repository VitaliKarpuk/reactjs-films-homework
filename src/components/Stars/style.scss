.stars__raiting {
  width: 300px;
  height: 50px;
  display: flex;
  justify-content: space-between;
  align-items: center;

  span {
    display: flex;
    align-items: center;
    width: 40px;
    height: 100%;
    font-size: 50px;
  }

  .star__rating_active {
    color: #0aaee4;
  }

  .star__rating_inactive {
    color: #ffffff;
  }

  .stars__raiting_value {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 60px;
    height: 36px;
    border-radius: 6px;
    background-color: transparent;
    border: 2px solid #0aaee4;
    font-size: 24px;
    line-height: 44px;
    color: #ffffff;
    font-weight: 500;
    margin-left: 30px;
  }
}
