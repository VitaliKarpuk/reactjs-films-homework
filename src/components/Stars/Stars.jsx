import React from "react";

import "./style.scss";

const Stars = () => {
  const arrNumberStars = [1, 2, 3, 4, 5];
  const value = 3;
  return (
    <div className="stars__raiting">
      {arrNumberStars.map((item, index) => {
        return index + 1 > value ? (
          <span key={item} className="star__rating_inactive">
            &#128970;
          </span>
        ) : (
          <span key={item} className="star__rating_active">
            &#128970;
          </span>
        );
      })}
      <div className="stars__raiting_value">{value}</div>
    </div>
  );
};

export default Stars;
