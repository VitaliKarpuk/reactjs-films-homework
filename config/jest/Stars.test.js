/* eslint-disable no-undef */
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Stars from '../../src/components/Stars/Stars';

describe('Test stars component', () => {
  test('renders correctly', () => {
    const renderer = new ShallowRenderer();
    const result = renderer.render(<Stars className='stars__raiting'/>);
    renderer.getRenderOutput();
    expect(result.type).toBe('div');
    expect(result.props.className).toBe('stars__raiting')
    expect(result.props.children[1].type).toBe('div');
    expect(result).toMatchSnapshot();
  })
})
