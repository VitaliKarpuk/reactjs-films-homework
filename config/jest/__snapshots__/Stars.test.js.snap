// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test stars component renders correctly 1`] = `
<div
  className="stars__raiting"
>
  <span
    className="star__rating_active"
  >
    ğŸŸŠ
  </span>
  <span
    className="star__rating_active"
  >
    ğŸŸŠ
  </span>
  <span
    className="star__rating_active"
  >
    ğŸŸŠ
  </span>
  <span
    className="star__rating_inactive"
  >
    ğŸŸŠ
  </span>
  <span
    className="star__rating_inactive"
  >
    ğŸŸŠ
  </span>
  <div
    className="stars__raiting_value"
  >
    3
  </div>
</div>
`;
